docs/src/tutorials/Add-New-Network.md

```diff
@@ -0,0 +1,164 @@
+
+```
+
+For networks supporting several tokens:
+
+```
+cargo run add_spec -u <network-ws-url> -<crypto> -token <decimals> <SYMBOL>
+
+```
+```
+// i.g.
+cargo run add_spec -u wss://karura-rpc-0.aca-api.network -sr25519 -token 12 KAR
+
+```
+
+Now your `<specs-file>` is in `parity-signer/rust/files/for_signing`
+
+
+### Sign Network Spec
+
+#### Get signature
+
+In `parity-signer/rust/files/for_signing`
+
+```
+cat <spec-file> | subkey sign --suri <seed-phrase>
+```
+
+```
+// i.g.
+cat sign_me_add_specs_statemint_sr25519 | subkey sign --suri "fluid upon nuclear level middle funny suffer author group shrimp tornado relax"
+```
+
+This will return a `<signature>` you need to make a signed QR.
+
+#### Make signed QR
+
+In `parity-signer/rust/generate_message`
+
+```
+cargo run make -qr -crypto <crypto> -msgtype add_specs -payload <spec-file> -verifier -hex <signature>
+```
+
+```
+// i.g.
+cargo run make -qr -crypto sr25519 -msgtype add_specs -payload sign_me_add_specs_statemint_sr25519 -verifier -hex 0x927c307614dba6ec42f84411cc1e93c6579893859ce5a7ac3d8c2fb1649d1542 -signature -hex fa3ed5e1156d3d51349cd9bb4257387d8e32d49861c0952eaff1c2d982332e13afa8856bb6dfc684263aa3570499e067d4d78ea2dfa7a9b85e8ea273d3a81a86
+```
+
+Now your `<spec-qr>` is in `parity-signer/rust/files/signed`
+
+### Feed Network Specs into Signer
+
+In Signer open scanner, scan your your `<spec-qr>` and approve chain specs.
+
+
+## Network Metadata
+
+### Get Network Metadata
+
+In `parity-signer/rust/generate_message`
+
+```
+cargo run load_metadata -d -u `<network-ws-url>`
+```
+
+```
+// i.g.
+cargo run load_metadata -d -u wss://statemint-rpc.polkadot.io
+```
+
+This will fetch fresh `<metadata-file>`, update the database with it, and - most relevant to us currently - generate file with message body in `parity-signer/rust/files/for_signing`. 
+
+### Sign Network Metadata
+
+#### Get Signature
+
+In `parity-signer/rust/files/for_signing`
+
+```
+cat <metadata-file> | subkey sign --suri <seed-phrase>
+```
+
+```
+// i.g.
+cat sign_me_load_metadata_statemintV800 | subkey sign --suri "fluid upon nuclear level middle funny suffer author group shrimp tornado relax"
+```
+
+#### Make signed QR
+
+In `parity-signer/rust/generate_message`
+
+```
+cargo run make -qr -crypto <crypto> -msgtype load_metadata -payload <metadata-file> -verifier -hex <signature>
+```
+
+```
+// i.g.
+cargo run make -qr -crypto sr25519 -msgtype load_metadata -payload sign_me_load_metadata_statemintV800 -verifier -hex 0x927c307614dba6ec42f84411cc1e93c6579893859ce5a7ac3d8c2fb1649d1542 -signature -hex 6a8f8dab854bec99bd8534102a964a4e71f4370683e7ff116c84d7e8d5cb344efd3b90d27059b7c8058f5c4a5230b792009c351a16c007237921bcae2ede2d84
+```
+
+This QR might take some time to be generated. After it is finished you can find your `<metadata-qr>` in `parity-signer/rust/files/signed`. It is a multipart QR-"movie", if you image viewer does not render it correctly, we suggest to open it in a browser.
```

Comment:
Try with `--release` key on a typical machine, I understand it's quite fast that way. Rust does too many checks on arithmetic in development mode and fountain QR generation is lots of additions.

Reply:
so here no need parity trusted certificate "0xc46a22b9...bf39bd57" for other network. is that parity-signer consider the first verifier in update command as the certificate for nondefault network like karura. will the  different network own different certificate?

Reply:
Every network can use independent certificate, except for those that have common *general certificate*. Parity trusted certificate could be replaced, after removing it the first certificate Signer accepts becomes general. Similarly, if general certificate is set, a certificate that is used for given genesis hash (that is, a network) is trusted on first use.

You do not need to remove general certificate to use other certificates for other networks, unless you want to. But if you do, main networks will be bound to general certificate.

The Parity general certificate exists only for convenience, of course decentralized approaches to updating are better, but since they require some effort, we offer simple (feature-poor yet secure) solution for those who only need basic features.